perf/src/packet.rs

```diff
@@ -81,37 +81,39 @@ impl PacketBatch {
     }
 }
 
-pub fn to_packet_batches<T: Serialize>(xs: &[T], chunks: usize) -> Vec<PacketBatch> {
-    xs.chunks(chunks)
-        .map(|x| {
-            let mut batch = PacketBatch::with_capacity(x.len());
-            batch.packets.resize(x.len(), Packet::default());
-            for (i, packet) in x.iter().zip(batch.packets.iter_mut()) {
-                Packet::populate_packet(packet, None, i).expect("serialize request");
+pub fn to_packet_batches<T: Serialize>(data: &[T], chunk_size: usize) -> Vec<PacketBatch> {
```

Comment:
How about `max_batch_size` instead of `chunk_size`?

Reply:
Also, `data` is usually used to describe a `&[u8]` type. Maybe could use `items`?

Reply:
`items` works for me; I used `chunk_size` to match the library variable name:
https://doc.rust-lang.org/src/core/slice/mod.rs.html#788

Reply:
cool, sounds good!